src/client/mod.rs

```diff
@@ -1,40 +1,63 @@
 //! Clients based on SRV lookups.
 
 use crate::{record::SrvRecord, resolver::SrvResolver};
-use arc_swap::{ArcSwap, Guard};
+use arc_swap::ArcSwap;
+use futures_util::{
+    pin_mut,
+    stream::{self, Stream, StreamExt},
+    FutureExt,
+};
 use http::uri::{Scheme, Uri};
-use std::{future::Future, ops::Deref, sync::Arc, time::Duration};
+use std::{
+    error::Error, fmt::Debug, future::Future, iter::FromIterator, sync::Arc, time::Duration,
+};
 
 mod cache;
 use cache::Cache;
 
 /// SRV target selection policies.
 pub mod policy;
-use policy::IntoUriIter;
 
 /// Errors encountered during SRV record resolution
 #[derive(Debug, thiserror::Error)]
-pub enum SrvError<Lookup: std::error::Error + 'static> {
+pub enum SrvError<Lookup: Debug> {
     /// Srv lookup errors
     #[error("srv lookup error")]
```

Comment:
I think merging this enum makes sense - the distinction was a little shaky.  Did make me notice that you're not consistent in the capitalization of the error strings...

Reply:
Fixed, thanks